lib/lint.ml

```diff
@@ -24,6 +24,27 @@ module Make (Docker : S.DOCKER_CONTEXT) = struct
     @@ install_ocamlformat
     @@ copy ~chown:"opam" ~src:["./"] ~dst:"./" ()
 
+  let odoc_dockerfile ~base ~opam_files =
+    let packages =
+      opam_files
+      |> Opam_build.packages_of_opam_files
+      |> String.concat " "
+    in
+    let opam_files =
+      opam_files
+      |> List.map ((^) "./")
+      |> String.concat " "
+    in
+    let open Dockerfile in
+    from (Docker.image_hash base)
+    @@ run "opam depext odoc"
+    @@ run "opam install dune odoc>=1.5.0"
+    (* Warnings-as-errors was introduced in ODoc.1.5.0 *)
+    @@ workdir "src"
+    @@ copy ~chown:"opam" ~src:["./"] ~dst:"./" ()
+    @@ run "opam depext --update %s" packages
+    @@ run "opam install --deps-only --yes %s" opam_files
```

Comment:
How will this work with the caching? It looks like it would force reinstalling all dependencies each each build.

Reply:
Indeed, that looks like a mistake :slightly_smiling_face: We could just make the `copy` step last?

Reply:
Yes, though you'll have to install just the opam files first so the install step works.

Reply:
Done in https://github.com/ocurrent/ocaml-ci/pull/105/commits/313a788fece55261f5bcb70a131ec58ceed94c76. I've done some local testing to confirm that this is caching the dependency installation appropriately.